keygen.py

```diff
@@ -1,73 +1,92 @@
+#!/usr/bin/env python3
+
 import string
 import random
-import sha
+import hashlib  # sha ~> hashlib.sha1
 
 BASE16 = "0123456789ABCDEF"
 BASE30 = "123456789ABCDEFGHJKLMNPQRTVWXY"
 
 
-def randomstring(size=20, chars=string.ascii_uppercase + string.digits):
+def random_string(size=20, chars=string.ascii_uppercase + string.digits):
     return "".join((random.choice(chars) for _ in range(size)))
```

Comment:
There's a `random.choices` function, it's more effective than `random.choice` multiple times